docs/proposals/odo-v3-cli/odo-describe-component.md

```diff
@@ -0,0 +1,35 @@
+
+# `odo describe component`
+
+Show detailed information about component.
+
+By default, it shows detail about component in the current directory.
+
+Alternatively users can use flags (`--component`, `--application`, `--namespace`) to specify existing component on the cluster and show information about it.
```

Comment:
Same request - please remove these flags. Please allow us to have a simpler odo v3 than its past iterations.

Reply:
can't how would user describe the component that exists on the cluster but for which there is no local devfile? This is critical functionality for IDE plugins.

Reply:
But why does it need to have these extra flags? How are IDE users going to know that there's a component on the cluster which they want to `odo describe`? That's going to happen with `odo list` - I don't know anything else. Then why can't we just `odo describe <component-name>` instead of taking these extra flags? The `--application` flag has really not served us any purpose. If they want to describe a component from a different project, what's stopping plugins from doing `odo project set <another-project>; odo describe <comp-name>` internally?

Reply:
My point is to keep odo layer simple and slim. Let's not imagine the scenarios that IDE plugins or other tooling needs and create flags/commands for that in odo. If the same thing can be done by running 2 odo commands, why do we want to add an extra command in odo and make it a maintenance overhead on odo?

Reply:
>But why does it need to have these extra flags? How are IDE users going to know that there's a component on the cluster which they want to odo describe

`odo list components`


>I don't know anything else. Then why can't we just odo describe <component-name> instead of taking these extra flags? 

that is not enough, component with the same name can be present in multiple applications

> If they want to describe a component from a different project, what's stopping plugins from doing odo project set <another-project>; odo describe <comp-name> internally?

`odo project set <another-project>;` would modify the currently active namespace in KUBECONFIG and it would affect all other `odo` `kubectl` or `oc` commands running at the same time.

Reply:
> that is not enough, component with the same name can be present in multiple applications

And I'm yet to see a sensible use of application flag happen. It's pretty easy for users to create different projects/namespaces to keep their stuff separate. Why are we still sticking with application label?



> `odo project set <another-project>;` would modify the currently active namespace in KUBECONFIG and it would affect all other `odo` `kubectl` or `oc` commands running at the same time.

I thought that IDE plugins were to use odo exclusively to communicate with a k8s/ocp cluster. If they are using `kubectl`/`oc` for other things, what's even the purpose of having odo for them? Why can't they do all communication using `kubectl`/`oc`? Rant apart, I see why `--namespace` flag could be useful at times. But `--component` and `--application` flag don't make sense to me. I would rather have them scrapped.

Reply:
> And I'm yet to see a sensible use of application flag happen. It's pretty easy for users to create different projects/namespaces to keep their stuff separate. Why are we still sticking with application label?

I've completely removed application support from alpha1


>I thought that IDE plugins were to use odo exclusively to communicate with a k8s/ocp cluster. If they are using kubectl/oc for other things, what's even the purpose of having odo for them? Why can't they do all communication using kubectl/oc? Rant apart, I see why --namespace flag could be useful at times. But --component and --application flag don't make sense to me. I would rather have them scrapped.

removed `--application` flag for now. Component needs to be there (I just renamed it to `--name`)